compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -439,7 +439,9 @@ pub fn maybe_create_entry_wrapper<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
 
         let isize_ty = cx.type_isize();
         let i8pp_ty = cx.type_ptr_to(cx.type_i8p());
+        let bool_ty = cx.type_i1();
         let (arg_argc, arg_argv) = get_argc_argv(cx, &mut bx);
+        let arg_ignore_sigpipe = bx.const_bool(!cx.tcx().features().no_ignore_sigpipe);
```

Comment:
This ignores `no_ignore_sigpipe` in library crates. Can we give a hard error instead that says SIGPIPE is a global resource that can only be set by the final binary?

Reply:
I addressed this particular comment in https://github.com/rust-lang/rust/pull/97802/commits/56301a1efc6c5df726406ebe238a98d933f019a4 plus added a test. There was no infrastructure in place for a check like this so I had to add that too. Feel free to review those particular parts of this PR already now.